lib/sort-order-selector/index.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import appState from '../flux/app-state';
+import {
+  toggleSearchSortOrder,
+  setSearchSortType,
+} from '../state/settings/actions';
+import classNames from 'classnames';
+import ArrowDownIcon from '../icons/arrow-down';
+
+export class SortOrderSelector extends Component {
+  static displayName = 'SortOrderSelector';
+
+  static propTypes = {
+    query: PropTypes.string.isRequired,
+    sortType: PropTypes.string.isRequired,
+    sortReversed: PropTypes.bool.isRequired,
+    searchSortType: PropTypes.string.isRequired,
+    searchSortReversed: PropTypes.bool.isRequired,
+    toggleSortOrder: PropTypes.func.isRequired,
+    setSortType: PropTypes.func.isRequired,
+  };
+
+  componentDidUpdate(prevProps) {
+    const { sortType, sortReversed, searchSortReversed, query } = this.props;
+    /* If there's a query and there wasn't before
+       (i.e., the search sort component is being newly displayed),
+       set the search sort to match the global setting.
+       This means whenever a user starts a new search, the notes will be filtered
+       in place, rather than being reordered to match the last selected search order. */
+    if (!prevProps.query && prevProps.query !== query) {
+      this.props.setSortType(sortType);
+      if (sortReversed !== searchSortReversed) {
+        this.props.toggleSortOrder();
+      }
+    }
+  }
+
+  changeSort = event => {
+    this.props.setSortType(event.currentTarget.value);
+  };
+
+  render() {
+    const { searchSortType, searchSortReversed, query } = this.props;
+
+    const sortTypes = [
+      {
+        label: 'Date modified',
+        id: 'modificationDate',
+      },
+      {
+        label: 'Date created',
+        id: 'creationDate',
+      },
+      {
+        label: 'Alphabetical',
+        id: 'alphabetical',
+      },
+    ];
+
+    return (
+      <Fragment>
+        {query.length > 0 && (
```

Comment:
If we leave this check for the calling parent component then it will clean up this function and we can remove the `<Fragment>`

It means moving the check to the parent component but at least then what it leaves us is a component that can stand on its own a bit better.